### PR TITLE
Check for missing values on contacts table

### PR DESCRIPTION
**Pull request by edmondsgarrett**
## Overview
Checks for empty values on registration contacts table
…

## Related

…

## Changes
- Checks if columns are defined for contacts table, otherwise defaults them to None
…

## Testing

1. In apps/admin_regis_table/views.py, replace lines 71-73 in `apps/admin_regis_table/views.py` with the following:
   <details><summary>Snippet</summary>
     
          registrations_content = [
            (
                1,                                  #registration_id
                '2013-01-23 10:49:36.880753',          #posted_date
                12023,                              #applicable_period_start
                122023,                             #applicable_period_end
                True,                               #submitting_for_self
                'received',                           #registration_status
                'carrier',                #org_type
                'BRule Insurance Company',    #business_name
                '7440 Woodland Drive',              #mail_address
                'Indianpolis',                      #city
                'IN',                               #state
                '46278     '                        #zip
            ),
            (
                2,                                  #registration_id
                '2023-01-23 10:49:36.880753',          #posted_date
                12023,                              #applicable_period_start
                122023,                             #applicable_period_end
                False,                               #submitting_for_self
                'processing',                           #registration_status
                'pbm',                #org_type
                'Clay Tenant Once Company',    #business_name
                '440 Wood and Drive',              #mail_address
                'Indianapolis',                      #city
                'NY',                               #state
                '24444     '                        #zip
            ),
            (
                3,                                  #registration_id
                '2023-01-23 12:49:36.880753',          #posted_date
                12023,                              #applicable_period_start
                122023,                             #applicable_period_end
                True,                               #submitting_for_self
                'complete',                        #registration_status
                'carrier',                #org_type
                'Magentation Insurance Company',    #business_name
                '742 Oddland Drive',              #mail_address
                'Indipolis',                      #city
                'MI',                               #state
                '62784     '                        #zip
            ),
            (
                1,                                  #registration_id
                '2023-05-03 10:49:36.880753',          #posted_date
                12023,                              #applicable_period_start
                122023,                             #applicable_period_end
                True,                               #submitting_for_self
                'received',                           #registration_status
                'tpa_aso',                #org_type
                'Golden Rule Insurance Co',    #business_name
                '7440 Woodland Drive',              #mail_address
                'Indianpolis',                      #city
                'IN',                               #state
                '46278     '                        #zip
            ),
            (
                2,                                  #registration_id
                '2022-01-23 10:49:36.880753',          #posted_date
                12023,                              #applicable_period_start
                122023,                             #applicable_period_end
                False,                               #submitting_for_self
                'processing',                           #registration_status
                'pbm',                #org_type
                'CTenant Insurance Company',    #business_name
                '440 Wood and Drive',              #mail_address
                'Indianapolis',                      #city
                'NY',                               #state
                '24444     '                        #zip
            ),
            (
                3,                                  #registration_id
                '2023-01-13 10:49:36.880753',          #posted_date
                12023,                              #applicable_period_start
                122023,                             #applicable_period_end
                True,                               #submitting_for_self
                'complete',                        #registration_status
                'tpa_aso',                #org_type
                'Magenta Conditn Insurance Company',    #business_name
                '742 Oddland Drive',              #mail_address
                'Indipolis',                      #city
                'MI',                               #state
                '62784     '                        #zip
            ),
            (
                1,                                  #registration_id
                '2003-03-03 10:49:36.880753',          #posted_date
                12023,                              #applicable_period_start
                122023,                             #applicable_period_end
                True,                               #submitting_for_self
                'received',                           #registration_status
                'carrier',                #org_type
                'Goden Rule Insurance Company',    #business_name
                '7440 Woodland Drive',              #mail_address
                'Indianpolis',                      #city
                'IN',                               #state
                '46278     '                        #zip
            ),
            (
                2,                                  #registration_id
                '2021-04-23 10:49:36.880753',          #posted_date
                12023,                              #applicable_period_start
                122023,                             #applicable_period_end
                False,                               #submitting_for_self
                'processing',                           #registration_status
                'pbm',                #org_type
                'Clay enant Insurance Company',    #business_name
                '440 Wood and Drive',              #mail_address
                'Indianapolis',                      #city
                'NY',                               #state
                '24444     '                        #zip
            ),
            (
                3,                                  #registration_id
                '2023-01-23 11:49:36.880753',          #posted_date
                12023,                              #applicable_period_start
                122023,                             #applicable_period_end
                True,                               #submitting_for_self
                'complete',                        #registration_status
                'carrier',                #org_type
                'Magenta Condition Insurance Company',    #business_name
                '742 Oddland Drive',              #mail_address
                'Indipolis',                      #city
                'MI',                               #state
                '62784     '                        #zip
            ),
            (
                1,                                  #registration_id
                '2009-11-23 10:49:36.880753',          #posted_date
                12023,                              #applicable_period_start
                122023,                             #applicable_period_end
                True,                               #submitting_for_self
                'received',                           #registration_status
                'carrier',                #org_type
                'Golden Rule Insurance Company',    #business_name
                '7440 Woodland Drive',              #mail_address
                'Indianpolis',                      #city
                'IN',                               #state
                '46278     '                        #zip
            ),
            (
                2,                                  #registration_id
                '2002-05-27 10:49:36.880753',          #posted_date
                12023,                              #applicable_period_start
                122023,                             #applicable_period_end
                False,                               #submitting_for_self
                'processing',                           #registration_status
                'pbm',                #org_type
                'Clay Tenant Insurance Company',    #business_name
                '440 Wood and Drive',              #mail_address
                'Indianapolis',                      #city
                'NY',                               #state
                '24444     '                        #zip
            ),
            (
                3,                                  #registration_id
                None,          #posted_date
                12023,                              #applicable_period_start
                122023,                             #applicable_period_end
                True,                               #submitting_for_self
                'complete',                        #registration_status
                'tpa_aso',                #org_type
                'Magenta Condition Insurance Company',    #business_name
                '742 Oddland Drive',              #mail_address
                'Indipolis',                      #city
                'MI',                               #state
                '62784     '                        #zip
            )
        ]
        registrations_entities = [
            (
                5, 1, 5, 46, 11111, None, 5, 'Garretts Test Business 2', '11-0001111', True, True, True, True, False
            ),
            (
                5, 2, 50000, 47, 11010, 21101, 1, 'A Second Test Entity', '00-0000000', True, False, False, True, False
            )
        ]
        registrations_contacts = [
            (
                52,
                1,
                False,
                'Test Role',
                'Garrett Test Tester',
                '',
                'notarealemail@emailplace.com'
            ),
            (
                53,
                1,
                False,
                '',
                '',
                '15555555555',
                'testemail@testemail.net'
            ),
            (
                54,
                2,
                True,
                'A 3rd and final test role',
                'Test 3rd Role Name Garrett',
                '0000000000',
                'email@gmail.com'
            )
        ]
    </details>

2. Go to http://localhost:8000/administration/list-registration-requests/ and confirm the page loads and all records show on table and in view/edit modals. Any missing contact info should instead display 'None'

## UI

…

### CODE REVIEW

**Review by sophia-massie — APPROVED** [review]
Tested locally - LGTM!

Some thoughts - can disregard:

Wanted to double check with you - aren't we safe with the role, name, and email since we're not running a function over them? It seems the error is coming from the format_phone_number function we put on the phone number. If the function doesn’t get what we initially expected it to, the function throws an error causing the 500 since you can't reverse a type None or null value. I think it would be sufficient to say only use this function on the response if there is one else return None. 
```
`def format_phone_number(num):
        formatted_num_list = list(reversed(num))
        last_four_digits = 3
        last_seven_digits = 7
        all_ten_digits = 11
        placement_corrector = 1 #To add formatting chars in correct places
        for curr_position in range(len(formatted_num_list) + 1):
            placement_position = curr_position + placement_corrector
            if curr_position == last_four_digits or curr_position == last_seven_digits:
                formatted_num_list.insert(placement_position,'-')
            if curr_position == all_ten_digits and len(num) > 10:
                formatted_num_list.insert(placement_position,' ')
                formatted_num_list.append('+')
        return ''.join(reversed(formatted_num_list))
    return {
...
    }`
```

File: apcd-cms/src/apps/utils/registrations_data_formatting.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def format_phone_number(num):
         'notif': reg_cont[2],
         'role': reg_cont[3],
         'name': reg_cont[4],
-        'phone': format_phone_number(reg_cont[5]),
+        'phone': format_phone_number(reg_cont[5]) if reg_cont[5] else None,
```
Comment (rstijerina) [inline apcd-cms/src/apps/utils/registrations_data_formatting.py:65]: So to confirm, `None` is an acceptable value for the `phone` field?

Comment (edmondsgarrett) [reply apcd-cms/src/apps/utils/registrations_data_formatting.py:65]: I'll check with UTH to confirm.

Comment (edmondsgarrett) [reply apcd-cms/src/apps/utils/registrations_data_formatting.py:65]: Confirmed with them, None is good!

**Review by rstijerina — APPROVED** [review]

**Pull request merged**